esp-hal/src/uart.rs

```diff
@@ -277,16 +277,14 @@ pub enum Error {
     ///
     /// This error occurs when the received data does not conform to the
     /// expected UART frame format.
-    #[allow(clippy::enum_variant_names, reason = "Frame error is a common term")]
-    RxFrameError,
+    RxFrame,
```

Comment:
If we're breaking this enum, we should probably update all variants to be consistent in terms of spelling, abbreviations, word order, and general descriptiveness.